### ISSUE TITLE
Extend custom Exceptions to be caught on parsing and evaluation

### ISSUE THREAD

**Issue by danvim** [ProcalApp/ProcalParser#1]
For API purposes, and to fully and clearly display all types of error messages, there is a need to throw custom errors on runtime during parsing and evaluation.

### Package planning

```
exceptions.* //where * are various custom exceptions by Procal
```

### Exceptions Catching

Exceptions will be caught in the `ProcalParser` Class, until future re-evaluation of the class.

```
... //evaluation code
}
catch (e: CustomException1) {...}
catch (e: CustomException2) {...}
... //rest of the exceptions
```

**Comment by mcreng** [comment]
Exception package implemented in commit 7cb75f0.

**Comment by mcreng** [comment]
Exception catching code structure added in commit 62b4935.
